helpers function to reconnect_switches lead to removal of all flows

Hi,

I was troubleshooting the issues that occurred in the e2e below:

```
rerun: 0
tests/test_e2e_20_flow_manager.py::TestE2EFlowManager::test_040_replace_action_flow: 2022-12-01,06:11:21.855076 - 2022-12-01,06:11:46.884125
self = <tests.test_e2e_20_flow_manager.TestE2EFlowManager object at 0x7f36d4a82220>

   def test_040_replace_action_flow(self):
     self.replace_action_flow()

tests/test_e2e_20_flow_manager.py:446: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.test_e2e_20_flow_manager.TestE2EFlowManager object at 0x7f36d4a82220>
restart_kytos = False

   def replace_action_flow(self, restart_kytos=False):

       payload = {
           "flows": [
               {
                   "priority": 10,
                   "idle_timeout": 360,
                   "hard_timeout": 1200,
                   "match": {
                       "in_port": 1
                   },
                   "actions": [
                       {
                           "action_type": "output",
                           "port": 2
                       }
                   ]
               }
           ]
       }

       api_url = KYTOS_API + '/flow_manager/v2/flows/00:00:00:00:00:00:00:01'
       requests.post(api_url, data=json.dumps(payload),
                     headers={'Content-type': 'application/json'})

       # wait for the flow to be installed
       time.sleep(10)

       # Verify the flow
       s1 = self.net.net.get('s1')
       flows_s1 = s1.dpctl('dump-flows')
       assert len(flows_s1.split('\r\n ')) == BASIC_FLOWS + 1, flows_s1
       assert 'in_port="s1-eth1' in flows_s1

       # Modify the actions and verify its modification
       s1.dpctl('mod-flows', 'actions=output:3')
       flows_s1 = s1.dpctl('dump-flows')
       assert 'actions=output:"s1-eth2"' not in flows_s1
       assert 'actions=output:"s1-eth3"' in flows_s1

       if restart_kytos:
           # restart controller keeping configuration
           self.net.start_controller(enable_all=True)
           self.net.wait_switches_connect()
       else:
           self.net.reconnect_switches()

       time.sleep(10)

       # Check that the flow keeps the original setting
       s1 = self.net.net.get('s1')
       flows_s1 = s1.dpctl('dump-flows')
     assert len(flows_s1.split('\r\n ')) == BASIC_FLOWS + 1, flows_s1
E       AssertionError:  cookie=0xab00000000000001, duration=14.026s, table=0, n_packets=8, n_bytes=336, priority=1000,dl_vlan=3799,dl_type=0x88cc actions=CONTROLLER:65535
E         
E       assert 1 == (3 + 1)
E        +  where 1 = len([' cookie=0xab00000000000001, duration=14.026s, table=0, n_packets=8, n_bytes=336, priority=1000,dl_vlan=3799,dl_type=0x88cc actions=CONTROLLER:65535\r\n'])
E        +    where [' cookie=0xab00000000000001, duration=14.026s, table=0, n_packets=8, n_bytes=336, priority=1000,dl_vlan=3799,dl_type=0x88cc actions=CONTROLLER:65535\r\n'] = <built-in method split of str object at 0x7f36b05bc100>('\r\n ')
E        +      where <built-in method split of str object at 0x7f36b05bc100> = ' cookie=0xab00000000000001, duration=14.026s, table=0, n_packets=8, n_bytes=336, priority=1000,dl_vlan=3799,dl_type=0x88cc actions=CONTROLLER:65535\r\n'.split

tests/test_e2e_20_flow_manager.py:441: AssertionError
```

One of the reasons why this error above happened was: once we ran the reconnect_switches function, all flows are removed.

See this discussion here: https://mail.openvswitch.org/pipermail/ovs-discuss/2014-May/033712.html

Proof of concept:

```
root@ee736f74a4a5:/kytos-end-to-end-tests# ovs-ofctl dump-flows s1
 cookie=0xac00000000000001, duration=26.644s, table=0, n_packets=0, n_bytes=0, priority=50000,dl_src=ee:ee:ee:ee:ee:02 actions=CONTROLLER:65535
 cookie=0xab00000000000001, duration=28.118s, table=0, n_packets=10, n_bytes=420, priority=1000,dl_vlan=3799,dl_type=0x88cc actions=CONTROLLER:65535
 cookie=0x0, duration=26.641s, table=0, n_packets=0, n_bytes=0, idle_timeout=360, hard_timeout=1200, priority=10,in_port="s1-eth1" actions=output:"s1-eth2"
root@ee736f74a4a5:/kytos-end-to-end-tests# ovs-vsctl del-controller s1; ovs-vsctl set-controller s1 tcp:127.0.0.1:6653; ovs-ofctl dump-flows s1
root@ee736f74a4a5:/kytos-end-to-end-tests# sleep 10
root@ee736f74a4a5:/kytos-end-to-end-tests# ovs-ofctl dump-flows s1
 cookie=0xac00000000000001, duration=19.088s, table=0, n_packets=0, n_bytes=0, priority=50000,dl_src=ee:ee:ee:ee:ee:02 actions=CONTROLLER:65535
 cookie=0xab00000000000001, duration=20.575s, table=0, n_packets=6, n_bytes=252, priority=1000,dl_vlan=3799,dl_type=0x88cc actions=CONTROLLER:65535
 cookie=0x0, duration=19.084s, table=0, n_packets=0, n_bytes=0, idle_timeout=360, hard_timeout=1200, priority=10,in_port="s1-eth1" actions=output:"s1-eth2"
```

The sleep shows that after some time the flows are recreated (due to the consistency check). The point is: right after changing the controller, the flows are removed.

I believe the expected behavior for reconnect_switches() is just to reset the connection to the controller (to force the consistency check to run). Removing the flows seems to be an unexpected behavior.

--- a/kytos-init.sh
+++ b/kytos-init.sh
@@ -4,7 +4,7 @@ set -xe
 
 # the settings below are intended to decrease the tests execution time (in fact, the time.sleep() calls
 # depend on the values below, otherwise many tests would fail)
-sed -i 's/STATS_INTERVAL = 60/STATS_INTERVAL = 30/g' /var/lib/kytos/napps/kytos/of_core/settings.py
+sed -i 's/STATS_INTERVAL = 60/STATS_INTERVAL = 7/g' /var/lib/kytos/napps/kytos/of_core/settings.py
 sed -i 's/CONSISTENCY_MIN_VERDICT_INTERVAL =.*/CONSISTENCY_MIN_VERDICT_INTERVAL = 60/g' /var/lib/kytos/napps/kytos/flow_manager/settings.py
 sed -i 's/LINK_UP_TIMER = 10/LINK_UP_TIMER = 1/g' /var/lib/kytos/napps/kytos/topology/settings.py
 sed -i 's/DEPLOY_EVCS_INTERVAL = 60/DEPLOY_EVCS_INTERVAL = 5/g' /var/lib/kytos/napps/kytos/mef_eline/settings.py

--- a/tests/helpers.py
+++ b/tests/helpers.py
@@ -325,11 +325,17 @@ class NetworkTest:
         self.start_controller(clean_config=True, enable_all=True)
         self.wait_switches_connect()
 
-    def reconnect_switches(self, target="tcp:127.0.0.1:6653"):
+    def reconnect_switches(self, target="tcp:127.0.0.1:6653",
+                           temp_target="tcp:127.0.0.1:6654"):
         """Restart switches connections.
-        This method can also be used to trigger a consistency check initial run."""
+        This method can also be used to trigger a consistency check initial run.
+
+        A temporary target is used in order to avoid OvS deleting the flows
+        if the controller config were to be deleted.
+        """
         for sw in self.net.switches:
-            sw.vsctl(f"del-controller {sw.name}")
+            sw.vsctl(f"set-controller {sw.name} {temp_target}")
+            sw.controllerUUIDs(update=True)
         for sw in self.net.switches:
             sw.vsctl(f"set-controller {sw.name} {target}")
             sw.controllerUUIDs(update=True)

